### PR TITLE
[improve][io] Improve kinesis connector config.

### PR DESCRIPTION
**Pull request by shibd**
### Motivation

Improve kinesis connector some config.

#### For the kinesis sink connector
1. Add `cloudwatchEndpoint` config,  Because the AWS Kinesis producer will periodically send log metrics to CloudWatch. 

The current user can set `awsRegion` or `awsEndpoint`, but if the user just sets `awsEndpoint`, he will encounter an error:
```
2023-08-08T21:16:36,307+0800 [kpl-daemon-0003] ERROR com.amazonaws.services.kinesis.producer.LogInputStreamReader - [2023-08-08 21:16:36.302919] [0x00007725][0x0000000201a76280] [error] [main.cc:268] Could not configure the region. It was not given in the config and we were unable to retrieve it from EC2 metadata.
2023-08-08T21:16:37,316+0800 [kpl-daemon-0000] ERROR com.amazonaws.services.kinesis.producer.KinesisProducer - Error in child process
com.amazonaws.services.kinesis.producer.IrrecoverableError: Child process exited with code 1
	at com.amazonaws.services.kinesis.producer.Daemon.fatalError(Daemon.java:537) ~[amazon-kinesis-producer-0.14.0.jar:?]
	at com.amazonaws.services.kinesis.producer.Daemon.fatalError(Daemon.java:509) ~[amazon-kinesis-producer-0.14.0.jar:?]
```
 

#### For the kinesis source connector
1. Change `applicationName` default value is `pulsar-kinesis`, If you use the original default values(`Apache Pulsar IO Connector`), you will encounter an error:
```
Caused by: software.amazon.awssdk.services.dynamodb.model.DynamoDbException: 1 validation error detected: Value 'Apache Pulsar IO Connector' at 'tableName' failed to satisfy constraint: Member must satisfy regular expression pattern: [a-zA-Z0-9_.-]+ (Service: DynamoDb, Status Code: 400, Request ID: FPG9PKE3V9TMEPUFBFFF7PGFEBVV4KQNSO5AEMVJF66Q9ASUAAJG)
	at software.amazon.awssdk.services.dynamodb.model.DynamoDbException$BuilderImpl.build(DynamoDbException.java:95) ~[dynamodb-2.10.56.jar:?]
	at software.amazon.awssdk.services.dynamodb.model.DynamoDbException$BuilderImpl.build(DynamoDbException.java:55) ~[dynamodb-2.10.56.jar:?]
```

#### For some refactor code:
1. Add a `KinesisSinkConfig.load()` method, and move the param verify logic to it.
2. Add a `KinesisSourceConfig.load()` method, and move the param verify logic to it.
3. Remove duplicate validation logic and useless tests
4. Remove `AwsCredentialProviderPlugin`, `AwsDefaultProviderChainPlugin` and `STSAssumeRoleProviderPlugin` class, These classes have long been deprecated, and there is no place to use them.

### Verifying this change
- Add `missCloudWatchEndpointTest` to cover `awsRegion` and `cloudwatchEndpoint` is empty.

### Documentation

- [ ] `doc` 
- [x] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository
- https://github.com/shibd/pulsar/pull/25

### CODE REVIEW

File: pulsar-io/kinesis/src/main/java/org/apache/pulsar/io/kinesis/KinesisSourceConfig.java [hunk 1]
```diff
@@ -122,24 +123,27 @@ public class KinesisSourceConfig extends BaseKinesisConfig implements Serializab
...
-    public static KinesisSourceConfig load(String yamlFile) throws IOException {
-        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
-        return mapper.readValue(new File(yamlFile), KinesisSourceConfig.class);
+    public static KinesisSourceConfig load(Map<String, Object> config, SourceContext sourceContext) {
+        KinesisSourceConfig kinesisSourceConfig = IOConfigUtils.loadWithSecrets(config,
+                KinesisSourceConfig.class, sourceContext);
+        checkArgument(isNotBlank(kinesisSourceConfig.getAwsRegion())
+                        || (isNotBlank(kinesisSourceConfig.getAwsEndpoint())
+                            && isNotBlank(kinesisSourceConfig.getCloudwatchEndpoint())
+                            && isNotBlank(kinesisSourceConfig.getDynamoEndpoint())),
```
[the remark this answers is not in the document]
Comment (shibd) [reply pulsar-io/kinesis/src/main/java/org/apache/pulsar/io/kinesis/KinesisSourceConfig.java:139]: Thanks for your suggestion. I create an `isNotBlankEndpoint` variable. This makes it easier to read a little.

File: pulsar-io/kinesis/src/main/java/org/apache/pulsar/io/kinesis/BaseKinesisConfig.java [hunk 2]
```diff
@@ -35,6 +35,13 @@ public abstract class BaseKinesisConfig implements Serializable {
     )
     private String awsEndpoint = "";
 
+    @FieldDoc(
+            required = false,
+            defaultValue = "",
+            help = "Cloudwatch end-point url. It can be found at "
+                    + "https://docs.aws.amazon.com/general/latest/gr/rande.html"
+    )
+    private String cloudwatchEndpoint = "";
```
Comment (Technoboy-) [inline pulsar-io/kinesis/src/main/java/org/apache/pulsar/io/kinesis/BaseKinesisConfig.java:44]: ·cloudWatchEndpoint` ？

Comment (shibd) [reply pulsar-io/kinesis/src/main/java/org/apache/pulsar/io/kinesis/BaseKinesisConfig.java:44]: https://github.com/apache/pulsar/pull/21004/files#diff-8cd0be28b901ed3ce6cebd6613bdc941f9748c0351f370573f1dd8cff32d00b7L130

To be consistent with the previous name.

**Review by Technoboy- — APPROVED** [review]

**Review by jiangpengcheng — APPROVED** [review]

**Comment by lhotari** [comment]
not cherry-picking since #24495 isn't required for 3.0.x anymore.

**Pull request merged**